services/core/java/com/android/server/locksettings/LockSettingsService.java

```diff
@@ -1732,6 +1732,142 @@ public boolean setLockCredential(LockscreenCredential credential,
         }
     }
 
+
+    private void enforceCallerPermisson() {
+        if (!hasPermission(PERMISSION) && !hasPermission(SET_AND_VERIFY_LOCKSCREEN_CREDENTIALS)) {
+            throw new SecurityException(
+                    "validDuressPinExist requires SET_AND_VERIFY_LOCKSCREEN_CREDENTIALS or "
+                            + PERMISSION);
+        }
+    }
+
+    @Override
+    public boolean validDuressCredentialsExist() {
+        enforceCallerPermisson();
+        return mStorage.readDuressPinToken() != null
```

Comment:
`readDuressPinToken` and `readDuressPasswordToken` are slow methods and their results can change between invocations.

As a minimum, they shouldn't be called twice. But better to use File.exists() checks instead